### PR TITLE
Support for clock buffer/mux modeling

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR adds a feature that allows to model clock buffers and multiplexers in V2X.
- Added new attribute `COMB_INCLUDE_CLOCKS`. When specified on a clock input port allows it to have combinational relations with other ports.
- Added a new attribute `NO_SEQ`. When set to 1 on an input port prevents V2X from attaching clock list to it.

### CODE REVIEW

**Comment by mkurc-ant** [comment]
@mithro I've gone through them. In my opinion for now only the `(* clock_signal = "{yes|no}" *)` matches the `CLOCK` attribute currently implemented in V2X.

We basically need most of the attributes to control XML generation an it is loosely coupled with actual synthesis/implementation control.

**Comment by mkurc-ant** [comment]
Requires https://github.com/SymbiFlow/python-symbiflow-v2x/pull/76 for the CI to pass.

File: v2x/vlog_to_model.py
Comment (mithro) [inline v2x/vlog_to_model.py:24]: Make sure there is a test for this.

File: v2x/vlog_to_model.py
Comment (mithro) [inline v2x/vlog_to_model.py:26]: Make sure there is a test for this.

File: v2x/vlog_to_model.py
Comment (mithro) [inline v2x/vlog_to_model.py:28]: Make sure there is a test for this.

File: v2x/vlog_to_model.py
Comment (mkurc-ant) [reply v2x/vlog_to_model.py:24]: It's under `tests/no_comb`

File: v2x/vlog_to_model.py
Comment (mkurc-ant) [reply v2x/vlog_to_model.py:26]: There is a test under `tests/clock_mux` that verifies that.

File: v2x/vlog_to_model.py
Comment (mkurc-ant) [reply v2x/vlog_to_model.py:28]: I've added a test under `tests/no_seq`

File: tests/no_seq/logic_macro.sim.v
Comment (mithro) [inline tests/no_seq/logic_macro.sim.v:111]: Weird indenting here....

**Comment by mithro** [comment]
@mkurc-ant - I think we can probably merge this after the docs update?

**Comment by mkurc-ant** [comment]
@mithro Yes. Once docs update is merged I'll add docs to this one.

**Comment by mkurc-ant** [comment]
@mithro I've rebased and updated docs.

File: tests/clock_mux/README.rst
Comment (mithro) [inline tests/clock_mux/README.rst:9]: As discussed on VC, the name is too confusing. I think we should rework this to maybe be a `(* COMB_INCLUDE_CLOCKS *)` attribute on the output?

File: tests/clock_mux/README.rst
Comment (mithro) [inline tests/clock_mux/README.rst:4]: Add something like this;
```
By default, clocks are excluded from the combinational output. IE They do not have `combinational_sink_ports` property associated with them.

However, clock multiplexers violate this rule.
```

File: tests/no_seq/README.rst
Comment (mithro) [inline tests/no_seq/README.rst:1]: I agree that this feature *probably* needed. I think you need to expand on *why* it is needed in human english. It also needs a significantly simpler example (rather than the *whole* quick logic logic cell) -- hopefully only 1-2 cells.

Comment (mkurc-ant) [reply tests/no_seq/README.rst:1]: I've reworked the example (simplified it).

File: tests/clock_mux/README.rst
Comment (mkurc-ant) [reply tests/clock_mux/README.rst:10]: That's a good idea. I've re-implemented the functionality so now there is the `(* COMB_INCLUDE_CLOCKS *)` attribute that has to be specified on an output port.

File: tests/clock_mux/README.rst
Comment (mkurc-ant) [reply tests/clock_mux/README.rst]: Done.

File: tests/no_seq/logic_macro.sim.v
Comment (mkurc-ant) [reply tests/no_seq/logic_macro.sim.v]: Fixed.

**Comment by mkurc-ant** [comment]
@mithro I've incorporated your requested changes. Could you re-review?

File: tests/clock_mux/gmux.sim.v
Comment (mithro) [inline tests/clock_mux/gmux.sim.v:34]: Too many new lines here.

**Review by mithro — APPROVED** [review]
LGTM!

File: tests/clock_mux/gmux.sim.v
Comment (mkurc-ant) [reply tests/clock_mux/gmux.sim.v:34]: Fixed.

**Comment by mkurc-ant** [comment]
@mithro Do you have any other questions to this PR or can we merge it?

**Comment by mithro** [comment]
Please update the pull request comment and then feel free to merge.

**Comment by mithro** [comment]
S/comment/description/

**Pull request merged**
